bundles/org.openhab.binding.sensorpush/src/main/java/org/openhab/binding/sensorpush/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,438 @@
+            ContentResponse authContentResponse = sendRequest(API_URL + Endpoint.AUTHORIZE, gson.toJson(authRequest),
+                    null);
+
+            if (authContentResponse == null) {
+                throw new InvalidResponseException("Auth content response is null");
+            }
+            int status = authContentResponse.getStatus();
+            if (status == 403) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid username/password");
+                return;
+            }
+            if (!goodStatus(authContentResponse.getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Auth request returned status " + Integer.valueOf(status));
+                return;
+            }
+            authResponse = gson.fromJson(authContentResponse.getContentAsString(), AuthorizationResponse.class);
+            if (authResponse == null) {
+                throw new InvalidResponseException("Auth response is null");
+            }
+            String authorizationToken = authResponse.authorization;
+            if (authorizationToken == null) {
+                throw new InvalidResponseException("Auth token is null");
+            }
+
+            // Send authorization token and get access token
+            logger.debug("Getting new access token");
+            String requestContent = gson.toJson(new AccessTokenRequest(authorizationToken));
+            ContentResponse accessContentResponse = sendRequest(API_URL + Endpoint.ACCESSTOKEN, requestContent, null);
+            if (accessContentResponse == null) {
+                throw new InvalidResponseException("Access token response is null");
+            }
+            if (!goodStatus(accessContentResponse.getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Access request returned status " + Integer.valueOf(status));
+                return;
+            }
+
+            tokenResponse = gson.fromJson(accessContentResponse.getContentAsString(), AccessTokenResponse.class);
+            if (tokenResponse == null) {
+                throw new InvalidResponseException("Access token object is null");
+            }
+            String accessToken = tokenResponse.accessToken;
+            logger.trace("Received access token: {}", accessToken);
+
+            JwtInfo jwtInfo;
+            try {
+                jwtInfo = new JwtInfo(accessToken);
+            } catch (IllegalArgumentException e) {
+                throw new InvalidResponseException("Invalid JWT token");
+            }
+            this.accessToken = accessToken;
+            accessTokenInfo = jwtInfo;
+        } catch (
+
+        TimeoutException e) {
+            logger.debug("Timeout sending auth request");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Auth request timeout");
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Reply:
Yes, it is logged by the framework. But to a different file than users never seem to include when reporting problems.

Reply:
I can imagine that users tend to forget to look into events.log, but that shouldn't be fixed in your binding. If that happens, you could ask them to include it.